### PR TITLE
Plot Regrets [Resolves #11]

### LINKED ISSUE #11: Add functionality for plotting results of selection rule regrets
@shaycrk we had talked about this a while ago. I think you said the functionality we want here would be a plot similar to the best-distance plot, but the distance being measured is the regret, and the lines would be selection rules instead of model groups. Is this accurate?

### PR DESCRIPTION
**Pull request by thcrock**
- Create SelectionRulePlotter that creates a cumulative distribution plot of selection rules and their regrets
- Implement selection rule grid to efficiently create many bound selection rules

### CODE REVIEW

File: audition/regrets.py [hunk 1]
```diff
@@ -55,13 +57,132 @@ def regrets_for_rule(
...
+        plot_cats(
+            frame=df_regrets,
+            x_col='regret',
+            y_col='pct_of_time',
+            cat_col=cat_col,
+            grp_col='selection_rule',
+            title=plt_title,
+            x_label='decrease in {} next time from best model'.format(regret_metric),
+            y_label='fraction of models',
+            x_ticks=numpy.arange(0, 1.1, 0.1)
```
Comment (shaycrk) [inline audition/regrets.py]: These ticks assume a 0-1 range, right? Also, the x-label of `decrease...` should probably be something like `distance from best`

**Comment by shaycrk** [comment]
Running through the code and poking at this in a notebook, looks pretty good to me.

File: audition/regrets.py (hunk 1 above)
Comment (thcrock) [reply audition/regrets.py]: Thanks, I just uploaded a change to use the same ticks that we already computed (as well as changing the copy)

**Comment by shaycrk** [comment]
Just getting around to testing this as well and noticed two things here:
* We should use the "distance from best" rather than "decrease in" language on the axis label for the plots looking at the model groups as well as the regret plots.
* Just using the same ticks as we're using to calculate the points for the plot is creating a tick density that's too high to be legible (see image below)

image

**Pull request merged**